packages/@lwc/ssr-compiler/src/__tests__/utils/expected-failures.ts

```diff
@@ -25,8 +25,6 @@ export const expectedFailures = new Set([
     'scoped-slots/mixed-with-light-dom-slots-outside/index.js',
     'slot-forwarding/slots/mixed/index.js',
     'slot-forwarding/slots/dangling/index.js',
-    'slot-not-at-top-level/advanced/lwcIf/light/index.js',
-    'slot-not-at-top-level/advanced/lwcIf/shadow/index.js',
     'slot-not-at-top-level/lwcIf-with-adjacent-text-nodes/light/index.js',
```

Comment:
Seems this new test will require `getLightSlottedContent` to also reuse the `irChildrenToEs` logic of setting `cxt.nextSibling`:

```diff
@@ -6,23 +6,23 @@
    <x-inner>
      <div>
        Inner!
      </div>
      <!---->
-     ‍‍
+     ‍‍‍
      <x-deep>
        <div>
          Deep!
        </div>
        <!---->
-       ‍
+       ‍
        <div>
          ‍
        </div>
-       ‍
+       ‍
        <!---->
      </x-deep>
-     ‍‍
+     ‍‍‍
      <!---->
    </x-inner>
    <!---->
  </x-outer>
```

Got the diff a bit smaller with a naive attempt setting in the traverse function:

```diff
@@ -6,11 +6,11 @@
    <x-inner>
      <div>
        Inner!
      </div>
      <!---->
-     ‍‍
+     ‍‍
      <x-deep>
        <div>
          Deep!
        </div>
        <!---->
@@ -19,10 +19,10 @@
          ‍
        </div>
        ‍
        <!---->
      </x-deep>
-     ‍‍
+     ‍‍
      <!---->
    </x-inner>
    <!---->
  </x-outer>
```

Not sure this PR should handle it though.

Reply:
I really appreciate all the effort you've put into this @cardoso! I'm going to spend a bit more time trying to break this PR (sorry, can't help it 😆) but at some point, I will timebox it, and we can move forward with this PR as-is since it is definitely an improvement. 🙂

I am still concerned with `cxt.isSlotted`, but I admit I don't have a better solution at the moment! 🤣

Reply:
I'm hoping that this test will be fixed by #5055. We'll probably have to combine the two PRs to more closely emulate what `flattenFragmentsInChildren()` does.

Reply:
OK upon testing the two PRs merged together, the issue here is that `lwc:if` should "reset" the adjacent text buffer. We need to render `<ZWJ><ZWJ>` rather than just `<ZWJ>` in this case:

```html
{empty}{empty}
        <template lwc:if={isTrue}>
            {empty}{empty}
```

We can handle this in a follow-up PR.

Reply:
- https://github.com/salesforce/lwc/pull/5056